### ISSUE TITLE
Bug - Icons - "cannot be used as a JSX component" error

### ISSUE THREAD

**Issue by mgrabovsky** [patternfly/patternfly-react#7353]
**Describe the problem**
When trying to use an icon inside a Patternfly component in TypeScript, I get a bunch of errors like the following:

    TS2786: 'OutlinedThumbsUpIcon' cannot be used as a JSX component.
      Its instance type 'Component<SVGIconProps, any, any>' is not a valid JSX element.
        89 |     return (
        90 |         <Button key="waived" variant="tertiary" onClick={onClick}>
      > 91 |             <OutlinedThumbsUpIcon />
           |              ^^^^^^^^^^^^^^^^^^^^
        92 |             <span className={styles.waive}>waive</span>
        93 |         </Button>
        94 |     );

**How do you reproduce the problem?**
I couldn't reproduce the error in CodeSandbox, but the [source code is open](https://github.com/fedora-ci/ciboard/blob/ae7bb4105a48fa640cd30291128529c85f10bb6d/src/components/ArtifactGreenwaveState.tsx#L91). Here is the PR where the error first occurred in a workflow run: https://github.com/fedora-ci/ciboard/pull/6

**Expected behavior**
Icons should be accepted as JSX components, I suppose?

**Is this issue blocking you?**
Not quite at the moment.

**What is your environment?**
GitHub workflow.

**Comment by jenny-s51** [comment]
Hi @mgrabovsky, thank you for bringing this to our attention! I made a CodeSandbox for this as well but could not reproduce, as this error is likely not related to Patternfly, but rather your build. I've linked a couple of resources below that should help address the TS error you are seeing! Within the past month it seems a lot of devs have been running into this recurring `react-types` issue.

- https://stackoverflow.com/questions/71791347/npm-package-cannot-be-used-as-a-jsx-component-type-errors/71828113#71828113
- https://github.com/facebook/react/issues/24304#issuecomment-1092563688 (same concept, but using your package manager)

Do either of these solutions work for you?
